### ISSUE TITLE
pkeyutl documentation on ECDSA wrong?

### ISSUE THREAD

**Issue by mouse07410** [openssl/openssl#8425]
Applies to OpenSSL-1.0.2r and OpenSSL-1.1.1-stable (haven't checked 1.1.1-dev).

### Problem

Man page for `openssl pkeyutl` is misleading/incorrect. It states:

> The EC algorithm supports sign, verify and derive operations. The sign and verify operations use ECDSA and derive uses ECDH. Currently there are no additional options other than digest. Only the SHA1 digest can be used and this digest is assumed by default.

Thankfully, ECDSA does support SHA256, SHA384, SHA512 - so the above text is wrong. 

Illustration:
```
$ openssl version
OpenSSL 1.0.2r  26 Feb 2019
$ openssl rand -out t.dat 48
$ openssl pkeyutl -engine pkcs11 -keyform engine -sign -inkey "pkcs11:manufacturer=piv_II;object=SIGN%20key;object-type=private" -pkeyopt digest:sha384 -in t.dat -out t.dat.sig
engine "pkcs11" set.
Enter PKCS#11 token PIN for Uri:
$ pkcs11-tool -r -y pubkey -d 02 > t-pub.der
Using slot 0 with a present token (0x0)
$ openssl pkeyutl -engine pkcs11 -keyform engine -verify -pubin -inkey "pkcs11:manufacturer=piv_II;object=SIGN%20pubkey;object-type=public" -pkeyopt digest:sha384 -in t.dat -sigfile t.dat.sig
engine "pkcs11" set.
Signature Verified Successfully
$ openssl dgst -engine pkcs11 -keyform engine -sign "pkcs11:manufacturer=piv_II;object=SIGN%20key;object-type=private" -sha384 -out t.dat.sig1 t.dat
engine "pkcs11" set.
Enter PKCS#11 token PIN for Uri:
$ openssl dgst -keyform DER -verify t-pub.der -sha384 -signature t.dat.sig1 t.dat
Verified OK
$ 
```

Documentation (`pkeyutl.1`) should be edited, removing from the `EC ALGORITHM` section the part about `SHA-1` digest, as it's clearly not the only one supported.

**Comment by mattcaswell** [comment]
Fix in #8432.

**Comment by mouse07410** [comment]
Thank you! Should this fix apply to 1.0.2 as well?

**Comment by mouse07410** [comment]
Closing, as the fix has been submitted, hopefully merged soon.
